core/shared/platform/common/posix/posix_socket.c

```diff
@@ -276,9 +276,10 @@ os_socket_recv_from(bh_socket_t socket, void *buf, unsigned int len, int flags,
             return -1;
         }
     }
-    else if (src_addr) {
-        memset(src_addr, 0, sizeof(*src_addr));
-    }
+    // This else if seem to overwrite src_addr when we don't want it to
```

Comment:
Same as above, if there is bug found in posix socket implementation, had better file an issue in the WAMR community, so others can check it and figure out how to resolve it.

Reply:
I investigated this case and found something, I will open an issue. In any case I restored the original implementation.